modules/Core/hooks/ValidateHook.php

```diff
@@ -14,17 +14,20 @@ public static function execute(UserValidatedEvent $event): void {
             define('VALIDATED_DEFAULT', 1);
         }
 
-        $validate_user = $event->user;
-        if (!$validate_user->exists()) {
-            return;
-        }
+        $validated_user = $event->user;
+
+        $groups_before = array_keys($validated_user->getGroups());
+
+        $validated_user->setGroup(VALIDATED_DEFAULT);
 
-        $validate_user->setGroup(VALIDATED_DEFAULT);
+        $groups_after = array_keys($validated_user->getGroups());
+        $groups_to_remove = array_diff($groups_before, $groups_after);
 
-        GroupSyncManager::getInstance()->broadcastChange(
-            $validate_user,
+        GroupSyncManager::getInstance()->broadcastGroupChange(
+            $validated_user,
             NamelessMCGroupSyncInjector::class,
-            [VALIDATED_DEFAULT]
+            [VALIDATED_DEFAULT],
```

Comment:
This still falsely triggering as user might already have the VALIDATED_DEFAULT group as many sites have Default and Validate group as same group

The old broadcastChange did this fine :S as it detected the real change in groups

Might just ignore this and lett go

Reply:
fixed